core/capabilities/ccip/launcher/launcher.go

```diff
@@ -189,195 +190,150 @@ func (l *launcher) processDiff(diff diffResult) error {
 	return err
 }
 
+// processUpdate will manage when configurations of an existing don are updated
+// If new oracles are needed, they are created and started. Old oens wil
 func (l *launcher) processUpdate(updated map[registrysyncer.DonID]registrysyncer.DON) error {
 	l.lock.Lock()
 	defer l.lock.Unlock()
 
 	for donID, don := range updated {
-		prevDeployment, ok := l.dons[registrysyncer.DonID(don.ID)]
+		prevPlugins, ok := l.instances[donID]
 		if !ok {
 			return fmt.Errorf("invariant violation: expected to find CCIP DON %d in the map of running deployments", don.ID)
 		}
-		// we encounter this when a node is removed from the don
-		if prevDeployment == nil {
-			return errors.New("this node was closed")
-		}
 
-		futDeployment, err := updateDON(
-			l.lggr,
-			l.myP2PID,
-			l.homeChainReader,
-			*prevDeployment,
-			don,
-			l.oracleCreator,
-		)
+		latestConfigs, err := getConfigsForDon(l.homeChainReader, don)
 		if err != nil {
 			return err
 		}
-		// When we remove a node from the don, this node does not have a future deployment
-		if futDeployment != nil {
-			if err := futDeployment.TransitionDeployment(prevDeployment); err != nil {
-				// TODO: how to handle a failed active-candidate deployment?
-				return fmt.Errorf("failed to handle active-candidate deployment for CCIP DON %d: %w", donID, err)
+
+		// If a config digest is not already in our list, we need to create an oracle
+		// If a config digest is already in our list, we just need to point to the old one
+		// newP.Transition will make sure we shut down the old oracles, and start the new ones
+		var newP ccipPlugins
```

Comment:
You should properly initialize the map to avoid the `if *c == nil` hack above

```suggestion
		newPlugins := make(ccipPlugins)
```